android/hybridcommon/src/main/java/com/revenuecat/purchases/hybridcommon/common.kt

```diff
@@ -611,8 +617,20 @@ fun purchaseSubscriptionOption(
             }
         }
 
+        val productIdsForAddOnStoreProducts = addOnStoreProducts?.mapNotNull {
+            val rawProductId = it["productIdentifier"] as? String ?: return@mapNotNull null
+            rawProductId.split(":").first()
+        }.orEmpty()
+        val productIdsForAddOnSubscriptionOptions = addOnSubscriptionOptions?.mapNotNull {
+            val rawProductId = it["productIdentifier"] as? String ?: return@mapNotNull null
+            rawProductId.split(":").first()
+        }.orEmpty()
+
+        val addOnProductsToFetch = productIdsForAddOnStoreProducts + productIdsForAddOnSubscriptionOptions
+        val productIdsToFetch = listOf(productIdentifier) + addOnProductsToFetch
```

Comment:
Maybe it's worth extracting to a separate function, something like `fetchProductsForAddOns`?